### ISSUE TITLE
[SPIKE] Decide about CentOS replacement or discontinuation

### ISSUE THREAD

**Issue by to-bar** [epiphany-platform/epiphany#2910]
**Is your spike related to a problem or idea? Please describe.**

> In December 2020, Red Hat terminated CentOS development. In response, CentOS founder Gregory Kurtzer created the Rocky Linux project as a successor to the original mission of CentOS. In March 2021, Cloud Linux (makers of CloudLinux OS) released a new RHEL derivative called AlmaLinux.
> 
> While CentOS was discontinued at the end of 2021, development of CentOS Stream, its midstream variant, continues.

_source_: https://en.wikipedia.org/wiki/CentOS

We are going to switch from RHEL 7 to 8 and because of this remove support for CentOS 7.

Some alternatives are described here:
https://thenewstack.io/where-to-turn-for-a-centos-replacement-heres-5-solid-linux-distros-to-check-out/

**Describe the outcome you'd like**
Comparison of distributions:
- AlmaLinux
- CentOS Stream
- Oracle Linux
- Rocky Linux 

**What is the reason or source for the spike**
CentOS Linux 8 discontinued at the end of 2021

**Additional context**
We support ARM architecture only with CentOS 7. 

---

**DoD checklist**

* [ ] Reader is able to understand the results of spike
* [ ] The results of the spike are presented in a table (to show simply what are compared or researched parameters) / not applicable
* [ ] Each value / cell in the results table is described more deeply below
* [ ] Demo of the spike (automated as much as possible)
* [ ] Design doc updated

**Comment by plirglo** [comment]
After discussions with teams in Epiphany 2.0 we don't need to support CentOS 8.
Continue work / research about alternatives listed in issue description (Rocky, Alma) to deliver system compatible with RHEL for developer purposes.

**Comment by to-bar** [comment]
## tl;dr
After comparison my recommendation is to choose AlmaLinux as replacement/successor to CentOS.
I find Rocky Linux as second choice option (as alternative).


**EL 8 distro comparison table:**
https://wiki.almalinux.org/Comparison.html

### 1) AlmaLinux

**Pros:**
- Fast releases (v8.5 three days after RHEL)
- 1:1 binary compatible with RHEL
- v8.x will be supported until 2029
- AlmaLinux OS Foundation created to take over AlmaLinux development and governance from CloudLinux
- $1M in annual sponsorship from CloudLinux Inc and support from other sponsors
- Wide list of mirrors (including PL), HiVelocity sponsorship (https://mirrors.almalinux.org/)
- Migration tool for CentOS 8.4 or 8.5 (https://github.com/AlmaLinux/almalinux-deploy)
- arm64 arch available on AWS 
- Official Vagrant boxes (4 providers)
- Commercial support available via TuxCare (a brand of CloudLinux)
- Very good and user friendly wiki (https://wiki.almalinux.org/)
- Official images on Docker Hub
- Tidy GitHub repos with roadmaps

**Additional info:**
- Owned by org type: Non-Profit 501(c)6

### 2) CentOS Stream

**Cons:**
- No 1:1 binary compatible with RHEL (upstream of RHEL)
- No 10 year lifecycle

### 3) Oracle Linux

**Cons:**
- By default it uses Oracle Unbreakable Enterprise Kernel (UEK) which contains some enhancements to make it more compatible with Oracle hardware and software (switching to Red Hat kernel is possible)
- No official ARM images (from Oracle) available on AWS (paid option from Supported Images available)

### 4) Rocky Linux

**Cons compared to AlmaLinux:**
- Release v8.5 three days later
- Modest wiki
- Not so tidy structure of Git repos
- Less providers for official Vagrant box (only 2 providers, 5 providers via `generic` by [Roboxes](https://roboxes.org/))
- No mirror in PL (https://mirrors.rockylinux.org/mirrormanager/mirrors)

**Additional info:**
- Owned by org type: For Profit, Public Benefit Corp
